src/Components/Table/CustomFooter.tsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import {
+  useGridApiContext,
+  useGridSelector,
+  gridPaginationModelSelector,
+  gridPageCountSelector,
+  gridPageSizeSelector,
+} from "@mui/x-data-grid";
+import {
+  Box,
+  FormControl,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  Typography,
+  Pagination,
+} from "@mui/material";
+
+function CustomPagination() {
```

Comment:
this is a react component: 
a. move it to a separate file named CustomPagination.tsx.
b. move this new componment and the CustomFooter file to a new folder named `components` inside of the Table folder.
c. after you move this component to the new folder, change the declaration to :
```
const CustomPagination = () => {
     ...everything that you wrote
}
```
instead of the regular function declaration.